dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/CheckUrlHelper.java

```diff
@@ -69,112 +63,47 @@ private static Optional<Boolean> checkUrl(String url, String baseURL) {
             }
             return Optional.empty();
         } catch (IOException e) {
+            LOGGER.error("Error checking url", e);
             return Optional.empty();
         } catch (InterruptedException e) {
             Thread.currentThread().interrupt();
             return Optional.empty();
         }
     }
 
-    private static Optional<Boolean> checkUrls(Set<String> urls, String baseURL) {
-        List<Optional<Boolean>> objectStream = urls.parallelStream().map(url -> checkUrl(url, baseURL))
-            .collect(Collectors.toCollection(ArrayList::new));
-        if (objectStream.stream().anyMatch(urlStatus -> urlStatus.isPresent() && urlStatus.get().equals(false))) {
-            return Optional.of(false);
-        }
-        if (objectStream.stream().anyMatch(Optional::isEmpty)) {
-            return Optional.empty();
-        }
-        return Optional.of(true);
+    private static boolean hasMalformedUrl(Set<String> possibleUrls) {
+        return possibleUrls.stream().anyMatch(possibleUrl -> {
+            try {
+                new URL(possibleUrl);
+            } catch (MalformedURLException e) {
+                LOGGER.debug("malformed url", e);
+                return true;
+            }
+            return false;
+        });
     }
 
     /**
-     * Get all the URLs from a JSON file.
-     *
-     * @param contents Contents of a JSON file
-     * @return The URLs
+     * Checks whether all <code>possibleUrls</code> are open access urls. Returns <code>Optional.of(true)</code>
```

Comment:
I should know this, but could use code comment. What makes a URL open access? Just that it doesn't require authentication?